refractor(stock-movement) improve seach modal

jeremielodi opened the pull request:
This PR solves this issue #2016

Closes #2016.

jeremielodi committed:
refractor(stock-movement) improve seach modal

mbayopanda commented on client/src/modules/stock/movements/registry.js:
```diff
@@ -248,11 +249,14 @@ function StockMovementsController(Stock, Notify,
+          vm.latestViewFilters = stockMovementFilters.formatView();
+
+          return load(stockMovementFilters.formatHTTP(true));
+        }
```
Could you adopt this style which is more clear
```js
if (!changes) { return; }
stockMovementFilters.replaceFilters(changes);
Stock.cacheFilters(filterKey);
vm.latestViewFilters = stockMovementFilters.formatView();

return load(stockMovementFilters.formatHTTP(true));
```

mbayopanda commented on server/controllers/stock/core.js:
```diff
@@ -61,17 +61,17 @@ function getLots(sqlQuery, parameters, finalClauseParameter) {
+        FROM lot l
+        JOIN inventory i ON i.uuid = l.inventory_uuid
+        JOIN inventory_unit iu ON iu.id = i.unit_id
+        JOIN stock_movement m ON m.lot_uuid = l.uuid AND m.flux_id in (${flux.FROM_PURCHASE})
```
Why do you change this equality like this ? 
This is code is about finding lots and their initial movement for lots which come from purchase, it was the first implementation.

Now initial lots come from : purchase, integration and donation, maybe you think about 
```js
... IN (${flux.FROM_PURCHASE}, ${flux.FROM_INTEGRATION}, ${flux.FROM_DONATION})
```

mbayopanda commented on client/src/modules/stock/movements/modals/search.modal.html:
```diff
@@ -60,9 +60,9 @@
+            <ui-select multiple name="inventory" ng-model="$ctrl.selectedFluxes">
               <ui-select-match>
-                <span translate>{{$select.selected.label}}</span>
+                <span translate>{{$item.label}}</span>
```
Could you try to set `append-to-body="true"` to the `ui-select` because the ui is not well.

![append to body](https://user-images.githubusercontent.com/5445251/30815500-521e7722-a20b-11e7-993b-0eff97d1e8dc.png)

jeremielodi replied:
the dropdow list disappear if I try to add append-to-body="true" to the ui-select

jniles committed:
fix(styles): allow ui-select overflow

This commit allows the `uiSelect` to overflow on top of a modal,
provided that "append-to-body" is given as _true_.  This fixes the
overflow issues in the stock modal.

For further information, see:
https://github.com/angular-ui/ui-select/wiki/FAQs#how-do-i-stop-the-drop-down-from-getting-cut-off-by-the-bottom-of-my-bootstrap-modal--container-with-overflow-hidden

jeremielodi committed:
Merge pull request #1 from jniles/refactor_stock_movement

fix(styles): allow ui-select overflow

jniles committed:
Merge branch 'master' into refractor_stock_movement
